src/jvmMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -180,6 +181,15 @@ internal class YamlOutput(
         return typeName
     }
 
+    private fun String.isAmbiguous(): Boolean {
+        return when {
```

Comment:
There are other patterns that are ambiguous - take a look at `YamlScalar`'s `convertToIntegerLikeValue`, `toFloat`, `toDouble` and `toBoolean` methods for most of the formats. 

`null`, `~` and empty strings will be interpreted as `null` - these need to be treated as ambiguous as well.